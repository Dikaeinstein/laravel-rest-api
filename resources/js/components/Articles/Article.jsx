import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Article = ({ article }) => (
  <div
    className='list-group-item list-group-item-action d-flex justify-content-between align-items-center'
    key={article.id}
    title="Click to view article detail"
  >
    {article.title}
    <Link
      className='btn btn-primary btn-sm mb-3'
      to={`articles/${article.id}`}
    >
      View Detail
    </Link>
  </div>
);

Article.propTypes = {
  article: PropTypes.objectOf(
    PropTypes.string.isRequired,
  ),
};

export default Article;
