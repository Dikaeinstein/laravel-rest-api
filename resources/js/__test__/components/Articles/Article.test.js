import React from 'react';
import { shallow } from 'enzyme';
import Article from '../../../components/Articles/Article';

describe('<Article />', () => {
  it('should render correctly', () => {
    const tree = shallow(<Article
      article={{
        id: '1',
        title: 'Test Article',
        body: 'This is a test article',
      }}
    />);
    expect(tree).toMatchSnapshot();
  });
});
