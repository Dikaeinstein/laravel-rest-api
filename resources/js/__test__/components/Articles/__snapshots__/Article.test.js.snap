// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> should render correctly 1`] = `
<div
  className="list-group-item list-group-item-action d-flex justify-content-between align-items-center"
  key="1"
  title="Click to view article detail"
>
  Test Article
  <Link
    className="btn btn-primary btn-sm mb-3"
    replace={false}
    to="articles/1"
  >
    View Detail
  </Link>
</div>
`;
